src/ui/mainwindow.cc

```diff
@@ -2851,135 +2847,66 @@ void MainWindow::hotKeyActivated( int hk )
   }
 }
 
-void MainWindow::prepareNewReleaseChecks()
-{
-  if ( cfg.preferences.checkForNewReleases )
-  {
-    QDateTime now = QDateTime::currentDateTime();
-
-    if ( !cfg.timeForNewReleaseCheck.isValid() ||
-         now.daysTo( cfg.timeForNewReleaseCheck ) > 2 )
-    {
-      // The date is invalid, or the check is set to happen more than 2 days
-      // in the future -- fix that.
-      cfg.timeForNewReleaseCheck = now.addDays( 2 );
-    }
-
-    int secsToCheck = now.secsTo( cfg.timeForNewReleaseCheck );
-
-    if ( secsToCheck < 1 )
-      secsToCheck = 1;
-
-    newReleaseCheckTimer.setSingleShot( true );
-    newReleaseCheckTimer.start( secsToCheck * 1000 );
-  }
-  else
-    newReleaseCheckTimer.stop(); // In case it was started before
-}
-
-void MainWindow::checkForNewRelease()
+void MainWindow::checkNewRelease()
 {
-  if ( latestReleaseReply ) {
-    disconnect( latestReleaseReply, 0, 0, 0 );
-    latestReleaseReply->deleteLater();
+  // Limit release check to 1 per day.
+  if ( cfg.timeForNewReleaseCheck < QDateTime::currentDateTime().addDays( 1 ) ) {
+    return;
   }
-  latestReleaseReply = 0;
 
-  QNetworkRequest req( QUrl( "https://github.com/xiaoyifang/goldendict/releases" ) );
+  QNetworkRequest github_release_api;
+  github_release_api.setUrl( QUrl( "https://api.github.com/repos/xiaoyifang/goldendict-ng/releases/latest" ) );
+  github_release_api.setRawHeader( QByteArray( "Accept" ), QByteArray( "application/vnd.github+json" ) );
+  // github_release_api.setRawHeader( QByteArray( "Authorization" ), QByteArray( "" ) );
+  github_release_api.setRawHeader( QByteArray( "X-GitHub-Api-Version" ), QByteArray( "2022-11-28" ) );
```

Comment:
does the github api  need token?

Reply:
and is there any way to check the pre-release version?

Reply:
> does the github api need token?

No, the rate limit without token is 60/hour which is more than enough.

> and is there any way to check the pre-release version?

Yes, there is a "list releases" api which will return recent 20 (or more) releases including the pre-releases.

But, at the current rate of at least 1 pre-release per 2 days from you. I don't think we should use pre-release :sweat_smile: 

https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#list-releases

Reply:
ok

Reply:
> No, the rate limit without token is 60/hour which is more than enough.

does this mean that if there are 100 users open the goldendict,only the first 60 will get the release check?

Reply:
No, the rate limit is for each IP.

> For unauthenticated requests, the rate limit allows for up to 60 requests per hour. Unauthenticated requests are associated with the originating IP address, and not the person making requests.

https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limits-for-requests-from-personal-accounts